pkg/controller/vpc.go

```diff
@@ -44,7 +44,7 @@ func (c *Controller) enqueueUpdateVpc(oldObj, newObj interface{}) {
 	oldVpc := oldObj.(*kubeovnv1.Vpc)
 	newVpc := newObj.(*kubeovnv1.Vpc)
 
-	if newVpc.DeletionTimestamp.IsZero() ||
+	if !newVpc.DeletionTimestamp.IsZero() ||
```

Comment:
How about just removing the !newVpc.DeletionTimestamp.IsZero(), I think the vpc `handleUpdate` has nothing to do with the delete logic.  all the clear logic should in the `handleDelVpc`

Reply:
Looking around, [the code contains the same pattern in multiple places](https://github.com/search?q=repo%3Akubeovn%2Fkube-ovn%20.DeletionTimestamp.IsZero()&type=code), like

https://github.com/kubeovn/kube-ovn/blob/60ef1ad1f31d2974cb9462670b0ad26b1015be6f/pkg/controller/qos_policy.go#L55-L59

https://github.com/kubeovn/kube-ovn/blob/60ef1ad1f31d2974cb9462670b0ad26b1015be6f/pkg/controller/subnet.go#L70-L73

So there might be a reason for why this check exists? Although I currently tend to agree and don't really see a reason for why you'd want to enqueue an update, when the `DeletionTimestamp` has been set, i.e. the VPC is scheduled for deletion at some point later in time.

Reply:
This pattern is aimed to prevent some race condition but I cannot recall the specific scenario. Maybe the controller receive an update event then  a deletion event, but it would process the deletion first then the update and lead to undesired garbage so we add a check before doing update. I think we can just keep the code like this as it can resolve this issue.